src/main.c

```diff
@@ -322,6 +325,56 @@ static NETX_CONSOLEAPP_RESULT_T opMode_erase(tFlasherInputParameter *ptAppParams
 /* ------------------------------------- */
 
 
+/**
+ * \brief opMode function for the smart erase feature.
+ *        Currently only supports SPI devices,
+ *        will fall back to normal erase for other devices.
+ * 
+ * \param ptAppParams  The flasher input parameters
+ * 
+ * \return NETX_CONSOLEAPP_RESULT_T
+ * - NETX_CONSOLEAPP_RESULT_OK     Flash was erase successfully
+ * - NETX_CONSOLEAPP_RESULT_ERROR  An error occured while erasing the flash
+ */
+static NETX_CONSOLEAPP_RESULT_T opMode_smartErase(tFlasherInputParameter *ptAppParams)
+{
+	CMD_PARAMETER_SMART_ERASE_T *ptParams = &(ptAppParams->uParameter.tSmartErase);
+
+	/* Be pessimistic. */
+	NETX_CONSOLEAPP_RESULT_T tResult = NETX_CONSOLEAPP_RESULT_ERROR;
+
+	/* get the source type */
+	BUS_T tSourceTyp = ptParams->ptDeviceDescription->tSourceTyp;
+	switch (tSourceTyp)
+	{
+	case BUS_SPI:
+		/*  use SPI flash */
+		tResult = spi_smart_erase(&(ptParams->ptDeviceDescription->uInfo.tSpiInfo), ptParams->ulStartAdr, ptParams->ulEndAdr);
+		if(tResult != 0){
+			uprintf("! smart_erase operation failed");
+			return tResult;
+		}
+		break;
+
+	case BUS_ParFlash:
+		/*  use parallel flash - not yet implemented.  */
+	case BUS_IFlash:
+		/*  use internal flash - not yet implemented.  */
+	case BUS_SDIO:
+		/*  use sdio flash - not yet implemented.  */
```

Comment:
If smart erase is not implemented, print a message "Smart erase is not implemented for parallel flash, using standard erase instead" and call normal erase.

Reply:
Sorry, I didn't see the call to erase in the default case :)